docs/current/cookbook.md

```diff
@@ -994,6 +994,40 @@ The following code listing demonstrates how to add multiple environment variable
 </TabItem>
 </Tabs>
 
+### Organize pipeline code into modules
+
+The following code listing demonstrates how to organize Dagger pipeline code into independent modules (or functions/packages, depending on your programming language) to improve code reusability and organization. It also demonstrates how to reuse the Dagger client and, therefore, share the Dagger session between modules.
```

Comment:
shall we add a note that users should follow this approach instead of creating new clients each time since that could potentially have issues? We should also clarify that the dagger client is concurrent safe. That's a question some users had in the past.

Reply:
Added note for first item. For the second, I think it should be documented in a separate location (Dagger CLI reference). I have opened https://github.com/dagger/dagger/issues/5729 for this.